src/Ng/Jobs/CatalogUtility.cs

```diff
@@ -79,9 +79,18 @@ public static HttpClient CreateHttpClient(bool verbose)
             var result = new SortedList<DateTime, IList<PackageDetails>>();
 
             XElement feed;
-            using (var stream = await client.GetStreamAsync(uri))
+            try
             {
-                feed = XElement.Load(stream);
+                using (var stream = await client.GetStreamAsync(uri))
+                {
+                    feed = XElement.Load(stream);
+                }
+            }
+            catch (TaskCanceledException tce)
+            {
+                // If the client gets disposed because the HTTP request timed out, a TaskCanceledException will be thrown.
+                // It is not necessary to restart the process in this case.
+                throw new TransientException("Http client was disposed.", tce);
```

Comment:
Adding resiliency for recoverable transient network conditions is the right thing to do.  However, do we know why HTTP GET requests are timing out after 5 minutes?  5 minutes is a long timeout.

Correct me if I'm wrong, but it looks like this spot is querying the OData feed for packages, while the [other place](https://github.com/NuGet/NuGet.Services.Metadata/pull/135/files#diff-368a08ab34918e5b2c6aee3991059d5eR186) where you catch a `TaskCanceledException` is querying CDN.  Rethrowing all `HttpClient` `TaskCanceledException`'s as a generic `TransientException` makes it difficult to understand the occurrence of faults in each location or distinguish a SQL timeout from an HTTP timeout; they're all transient and potentially recoverable, but you'd have to crawl exception call stacks or parse free-form exception messages to begin to perform root cause analysis.  If you can add more context at the time of failure to improve future monitoring and alerting, it will help down the road.

One option is to subclass `TransientException` with `HttpGetPackagesFailedException` and `HttpGetPackageMetadataFailedException`.  Then log the full exception (including its type).